splice_machine/src/test/java/com/splicemachine/subquery/Subquery_Flattening_Aggregate_IT.java

```diff
@@ -660,6 +663,51 @@ public void unsupported_nestedCorrelatedThreeLevels() throws Exception {
                 " 5 |50 |500 |");
     }
 
+    @Test
+    public void unsupported_value_producing_aggregate() throws Exception {
+        String sql = "select col1, col2 from t1 where (select count(*) from t2 where col1 = t1.col1) = 0";
+        assertUnorderedResult(sql, 1, "" +
+                "COL1 |COL2 |\n" +
+                "------------\n" +
+                " 42  | 43  |");
+
+        sql = "select col1, col2 from t1 where (select count(*)+10 from t2 where col1 = t1.col1) = 10";
+        assertUnorderedResult(sql, 1, "" +
+                "COL1 |COL2 |\n" +
+                "------------\n" +
+                " 42  | 43  |");
+
+        sql = "select col1, col2 from t1 where (select count(t1.col1)+10 from t2 where col1 = t1.col1) = 10";
```

Comment:
can you add some tests that check the flattening is still happening for the valid cases? Or is that already covered by some other ITs?

Reply:
done